be/src/storage/dictionary_cache_manager.h

```diff
@@ -21,32 +21,48 @@
 #include "column/chunk.h"
 #include "column/column.h"
 #include "column/datum.h"
+#include "common/compiler_util.h"
 #include "common/status.h"
 #include "exec/dictionary_cache_writer.h"
 #include "fmt/format.h"
 #include "service/internal_service.h"
 #include "storage/chunk_helper.h"
 #include "storage/primary_key_encoder.h"
-#include "storage/row_store_encoder_factory.h"
-#include "storage/row_store_encoder_simple.h"
-#include "util/faststring.h"
+#include "storage/type_traits.h"
 #include "util/phmap/phmap.h"
 #include "util/xxh3.h"
 
 namespace starrocks {
 
+#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))
+#include <mmintrin.h>
+#define PREFETCH_ADDR(addr) _mm_prefetch((const char*)(addr), _MM_HINT_T0)
+#elif defined(__GNUC__)
+#define PREFETCH_ADDR(addr) __builtin_prefetch(static_cast<const void*>(addr), 0 /* rw==read */, 3 /* locality */)
+#endif // __GNUC__
+
 enum DictionaryCacheEncoderType {
-    ROW_STORE_SIMPLE = 0,
+    PK_ENCODE = 0,
```

Comment:
Will there be any compatibility issues when upgrading BE?

Reply:
I think it will not have any  compatibility issues. Because dictionary cache is in-memory data structure, it is safe to do it because upgrading need to restart BE.